pkg/store/bucket.go

```diff
@@ -284,7 +287,19 @@ func (s *BucketStore) SyncBlocks(ctx context.Context) error {
 	close(blockc)
 	wg.Wait()
 
-	return err
+	if err != nil {
+		return err
+	}
+	// Drop all blocks that are no longer present in the bucket.
+	for id := range s.blocks {
+		if _, ok := allIDs[id]; ok {
+			continue
+		}
+		if err := s.removeBlock(id); err != nil {
```

Comment:
wow, lot's of races here (: during compaction

Reply:
but I guess we just end-up having maybe weird results during compaction on query level.

Reply:
Well, for the duration a new block and its sources exist concurrently, nothing weird should be showing. It's the same data and we should have measures in place to skip over those.

What could happen is that we delete old blocks before properly seeing their replacement. That should be rare and can be addressed if it occurs in practice.